sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java

```diff
@@ -95,6 +98,11 @@ private void scanFile(InputFile inputFile) {
       AstNode astNode = parser.parse(pythonFile.content());
       PyFileInputTree parse = new PythonTreeMaker().fileInput(astNode);
       visitorContext = new PythonVisitorContext(astNode, parse, pythonFile);
+      List<PythonSubscriptionCheck> checksBasedOnTree = checks.all().stream()
```

Comment:
Why not do the instance of test in the loop a few lines down ? You don't need to pass the "parse" tree.